extensions/sql/data/MaxCollector.h

```diff
@@ -0,0 +1,147 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <string>
+#include <unordered_map>
+#include <tuple>
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace sql {
+
+struct Date : public std::string {
+  Date(const std::string& str)
+    : std::string(str) {
```

Comment:
What is this class supposed to do?

Reply:
Idea is to differentiate date from string for comparison, but in MaxCollector date is formatted as 'Y-m-d H:M:S.999' so string comparison works correctly, there is a comment about it for `valueGreaterMaxValue(const Date& value, const Date& maxValue)`. So technically can be used string instead.